src/Dev/BuildTask.php

```diff
@@ -2,103 +2,112 @@
 
 namespace SilverStripe\Dev;
 
-use SilverStripe\Control\HTTPRequest;
-use SilverStripe\Core\Config\Config;
-use SilverStripe\Core\Config\Configurable;
+use LogicException;
 use SilverStripe\Core\Extensible;
-use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\PolyExecution\PolyCommand;
+use SilverStripe\PolyExecution\PolyOutput;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
 
 /**
- * Interface for a generic build task. Does not support dependencies. This will simply
- * run a chunk of code when called.
- *
- * To disable the task (in the case of potentially destructive updates or deletes), declare
- * the $Disabled property on the subclass.
+ * A task that can be run either from the CLI or via an HTTP request.
+ * This is often used for post-deployment tasks, e.g. migrating data to fit a new schema.
  */
-abstract class BuildTask
+abstract class BuildTask extends PolyCommand
 {
-    use Injectable;
-    use Configurable;
     use Extensible;
 
-    public function __construct()
-    {
-    }
-
     /**
-     * Set a custom url segment (to follow dev/tasks/)
-     *
-     * @config
-     * @var string
-     * @deprecated 5.4.0 Will be replaced with $commandName
+     * Shown in the overview on the {@link TaskRunner}
+     * HTML or CLI interface. Should be short and concise.
+     * Do not use HTML markup.
      */
-    private static $segment = null;
+    protected string $title;
 
     /**
-     * Make this non-nullable and change this to `bool` in CMS6 with a value of `true`
-     * @var bool|null
+     * Whether the task is allowed to be run or not.
+     * This property overrides `can_run_in_cli` and `can_run_in_browser` if set to false.
      */
-    private static ?bool $is_enabled = null;
+    private static bool $is_enabled = true;
 
     /**
-     * @var bool $enabled If set to FALSE, keep it from showing in the list
-     * and from being executable through URL or CLI.
-     * @deprecated - remove in CMS 6 and rely on $is_enabled instead
+     * Describe the implications the task has, and the changes it makes.
+     * Do not use HTML markup.
      */
-    protected $enabled = true;
+    protected static string $description = 'No description available';
 
-    /**
-     * @var string $title Shown in the overview on the {@link TaskRunner}
-     * HTML or CLI interface. Should be short and concise, no HTML allowed.
-     */
-    protected $title;
+    private static array $permissions_for_browser_execution = [
+        'ADMIN',
+        'anyone_with_dev_admin_permissions' => 'ALL_DEV_ADMIN',
+        'anyone_with_task_permissions' => 'BUILDTASK_CAN_RUN',
```

Comment:
```suggestion
        'ALL_DEV_ADMIN',
        'BUILDTASK_CAN_RUN',
```
I found the additional of 'helpful' keys just confusing, I thought they were adding some additional functionality when they're not

Reply:
They _are_ adding additional functionality - you can set these to `null` in subclasses if you don't want people with those permissions to access a specific task. You might have a task that's highly destructive that only admins can run, for example.